src/components/app/data/hooks/useHasAvailableSubsidiesOrRequests.js

```diff
@@ -0,0 +1,69 @@
+import { useMemo } from 'react';
+import { getOfferExpiringFirst, getPolicyExpiringFirst } from '../../../dashboard/sidebar/utils';
+import { LICENSE_STATUS } from '../../../enterprise-user-subsidy/data/constants';
+import useBrowseAndRequest from './useBrowseAndRequest';
+import useCouponCodes from './useCouponCodes';
+import useEnterpriseOffers from './useEnterpriseOffers';
+import useRedeemablePolicies from './useRedeemablePolicies';
+import useSubscriptions from './useSubscriptions';
+
+function getLearnerCreditSummaryCardData({ enterpriseOffers, redeemableLearnerCreditPolicies }) {
+  const learnerCreditPolicyExpiringFirst = getPolicyExpiringFirst(redeemableLearnerCreditPolicies?.redeemablePolicies);
+  const enterpriseOfferExpiringFirst = getOfferExpiringFirst(enterpriseOffers);
+
+  if (!learnerCreditPolicyExpiringFirst && !enterpriseOfferExpiringFirst) {
+    return undefined;
+  }
+
+  return {
+    expirationDate: (
+      learnerCreditPolicyExpiringFirst?.subsidyExpirationDate || enterpriseOfferExpiringFirst?.endDatetime
+    ),
+  };
+}
+
+/**
+ *
+ * @returns {{
+ * hasAvailableLearnerCreditPolicies: boolean,
+ * hasAssignedCodesOrCodeRequests: boolean,
+ * hasActiveLicenseOrLicenseRequest: boolean,
+ * learnerCreditSummaryCardData: {expirationDate},
+ * hasAvailableSubsidyOrRequests: ({expirationDate}|boolean)
+ * }}
+ */
+export default function useHasAvailableSubsidiesOrRequests() {
```

Comment:
Now that this logic is abstracted out as a hook, it's no longer covered by any unit tests (per the codecov annotations) due to mocking in the components. Given the added, important business logic (and the prior coverage), we should ensure this hook is covered by independent unit tests now.